client/src/Annotator/Info/useFilter.tsx

```diff
@@ -0,0 +1,38 @@
+import { useState, useMemo, useCallback } from 'react';
+
+import { Category } from '../../common/types';
+
+interface IUseFilter {
+    (categories: Category[]): UseFilterResponse;
+}
+interface UseFilterResponse {
+    searchText: string;
+    filteredIds: number[];
+    setSearchText: (val: string) => void;
+}
+
+const useFilter: IUseFilter = categories => {
+    const [searchText, _setSearchText] = useState<string>('');
+
+    const filteredIds: number[] = useMemo(() => {
+        if (searchText.length <= 2) {
+            return categories.map(o => o.id);
+        }
+        const search = searchText.toLowerCase();
+        const cats = categories.filter(category =>
```

Comment:
![](http://becomeawag.com/wp-content/uploads/2019/09/coughing-cat-know-your-meme-6.png)

Please change the variable name